public/lessonBuilder/lessonItemsController.js

```diff
@@ -97,28 +62,95 @@ angular.module('newApp')
         }
     }
 
+    function findTeacherRes(modules, moduleName) {
+        for(i = 0; i < modules.length; i++) {
+            // find the module tha has moduleName AND "Teacher" in it.
+            if(modules[i].name.search(moduleName) != -1) {
+                if(modules[i].name.search("Teacher") != -1)
```

Comment:
Instead of having two if statements here, they could be merged into a compound conditional.
`if (modules[i].name.search(moduleName) !== -1 && modules[i].name.search("Teacher") !== -1) {`

Also, I'm curious, why did you use `search()` instead of `indexOf()`?